src/Extensions/Logging/Internal/Replayable/OmexLogEventReplayer.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Logging.Internal.Replayable;
+using Microsoft.Omex.Extensions.Logging.Replayable;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	internal class OmexLogEventReplayer : ILogEventReplayer
+	{
+		private readonly ILogEventSender m_sender;
+		private readonly IOptionsMonitor<OmexLoggingOptions> m_options;
+
+		public OmexLogEventReplayer(ILogEventSender sender, IOptionsMonitor<OmexLoggingOptions> options)
+		{
+			m_sender = sender;
+			m_options = options;
+		}
+
+		public bool IsReplayableMessage(LogLevel logLevel) =>
+			logLevel switch
+			{
+				LogLevel.Trace or LogLevel.Debug => m_options.CurrentValue.ReplayLogsInCaseOfError,
+				_ => false
+			};
+
+		public void AddReplayLog(Activity activity, LogMessageInformation logMessage) =>
+			activity.AddReplayLog(logMessage, m_options.CurrentValue.MaxReplayedEventsPerActivity);
+
+		public void ReplayLogs(Activity activity)
+		{
+			// Replay parent activity
+			if (activity.Parent != null)
+			{
+				ReplayLogs(activity.Parent);
+			}
+
+			foreach (LogMessageInformation log in activity.GetReplayableLogs())
+			{
+				m_sender.LogMessage(activity, log.Category, LogLevel.Information, log.EventId, log.ThreadId, log.Message, log.Exception);
```

Comment:
>Message [](start = 101, length = 7)

The replayed logs from the legacy system used to be serialized into a JSON with replay tag, it was easy to differentiate them. Do we have something similar here or we just replay the raw logs? Not being able to differentiate them can cause confusion if we hit the max replay limit.

Reply:
I think it's useful to keep tags and categories to simplify the search, but it's a good very point that they should be distinguishable so I've added a prefix to the message text.